src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx

```diff
@@ -85,33 +85,32 @@ const UserVocabularyProcessScreen = ({ navigation }: UserVocabularyProcessScreen
       return
     }
 
-    await getNextUserVocabularyId().then(id => {
-      incrementNextUserVocabularyId()
-        .then(() => {
-          const imagePaths: Images = []
-          images.forEach(async (image, index) => {
-            const path = `${DocumentDirectoryPath}/image-${id}-${index}.txt`
-            imagePaths.push({ id: index, image: path })
-            await writeFile(path, image, 'utf8')
-          })
-
-          addUserDocument({
-            id,
-            word,
-            article: ARTICLES[articleId],
-            document_image: imagePaths,
-            audio: '',
-            alternatives: [],
-          })
-            .then(() => navigation.navigate('UserVocabularyList', { headerBackLabel: getLabels().general.back }))
-            .finally(() => {
-              setWord('')
-              setArticleId(null)
-              setImages([])
-            })
-        })
+    const id = await getNextUserVocabularyId()
+    await incrementNextUserVocabularyId()
+
+    const imagePaths: Images = []
+    images.forEach(image => {
+      const index = images.indexOf(image)
```

Comment:
Out of curiosity, why the change from using the index from forEach to this?

Reply:
hmh, no idea. Undid this.